### ISSUE TITLE
Why each object's color could be represented as a 3x4 channel tensor in training stage 1

### ISSUE THREAD

**Issue by iris0329** [Chat-3D/Chat-3D-v2#18]
Hi, Thanks for sharing this awesome work, it's super insightful.

I found that to get the attribute-aware token embedding, the color information is needed.

But I wonder why each object's `color` could be represented as a `3x4 channel tensor` in training stage 1? What does it mean, and how did you get this color tensor?

I would appreciate a lot of your response.

```python
class Chat3D(nn.Module):
    ...
    def encode_object_feat(self, feat, locs, colors):
        # feat = self.object_input_proj(feat)
        size_emb = self.coord_proj(locs[:, :, 3:6]) # [bs, 1, 512]
        gmm_weights = colors[..., :1]
        gmm_means = colors[..., 1:]
        gmm_colors = torch.sum(gmm_weights * gmm_means, dim=2) # [bs, 1, 3]
        # color_emb = self.color_dropout(torch.sum(self.color_proj(gmm_means) * gmm_weights, dim=2))
        color_emb = self.color_proj(gmm_colors)
        feat = torch.cat([feat, size_emb, color_emb], dim=-1)
        # feat = torch.cat([feat, size_emb], dim=-1)
        # feat = self.scene_proj(feat)
        return feat
```

**Comment by ZzZZCHS** [comment]
For the meaning of the 3x4 tensor, you can refer to this [issue](https://github.com/Chat-3D/Chat-3D-v2/issues/9#issuecomment-1908181349).
We followed [vil3dref](https://github.com/cshizhe/vil3dref/tree/main/preprocess/scannetv2)'s code to prepare scannet data, which results in these 3x4 color tensors. Actually, in our model, `gmm_colors = torch.sum(gmm_weights * gmm_means, dim=2)` transforms it back to a 1x3 color tensor, which theoretically should be equal to the mean value of this instance's point cloud colors.

**Comment by iris0329** [comment]
Ah, I see, thanks for your reply
